skimage/measure/_structural_similarity.py

```diff
@@ -3,36 +3,137 @@
+        for d in range(X.ndim):
+            slice_array.append(slice(pad[d], -pad[d]))
+
+    return X[slice_array]
+
+
+def structural_similarity(X, Y, win_size=None, gradient=False,
+                          dynamic_range=None, multichannel=None,
+                          gaussian_weights=False, full=False,
+                          image_content_weighting=False, **kwargs):
     """Compute the mean structural similarity index between two images.
 
     Parameters
     ----------
-    X, Y : (N,N) ndarray
+    X, Y : ndarray
         Images.
-    win_size : int
-        The side-length of the sliding window used in comparison.  Must
-        be an odd value.
+    win_size : int or None
+        The side-length of the sliding window used in comparison.  Must be an
+        odd value.  Default is 11 if `gaussian_weights` is True, 7 otherwise.
     gradient : bool
         If True, also return the gradient.
     dynamic_range : int
-        Dynamic range of the input image (distance between minimum and
-        maximum possible values).  By default, this is estimated from
-        the image data-type.
+        Dynamic range of the input image (distance between minimum and maximum
+        possible values).  By default, this is estimated from the image
+        data-type.
+    multichannel : int or None
+        If True, treat the last dimension of the array as channels. Similarity
+        calculations are done independently for each channel then averaged.
+        Defaults to True only if X is 3D and X.shape[2] == 3.
+    gaussian_weights : bool
+        If True, each patch (of size win_size) has its mean and variance
+        spatially weighted by a normalized Gaussian kernel of width sigma=1.5.
+    full : bool
+        If True, return the full structural similarity image instead of the
+        mean value
+    image_content_weighting : bool
+        If True, weight the ssim mean is spatially weighted by image content as
+        proposed in Wang and Shang 2006 [3].
+
+    Other Parameters
+    ----------------
+    use_sample_covariance : bool
+        if True, normalize covariances by N-1 rather than, N where N is the
+        number of pixels within the sliding window.
+    K1 : float
+        algorithm parameter, K1 (small constant, see [1])
+    K2 : float
+        algorithm parameter, K2 (small constant, see [1])
+    sigma : float
+        sigma for the Gaussian when `gaussian_weights` is True.
 
     Returns
     -------
-    s : float
-        Structural similarity.
-    grad : (N * N,) ndarray
-        Gradient of the structural similarity index between X and Y.
-        This is only returned if `gradient` is set to True.
+    mssim : float or ndarray
+        mean structural similarity.
+    grad : ndarray
+        Gradient of the structural similarity index between X and Y [2]. This
+        is only returned if `gradient` is set to True.
+    S : ndarray
+        Full SSIM image.  This is only returned if `full` is set to True.
```

Comment:
I guess I would say: keep the returns constant if you get all the
outputs anyway.  If you have to do additional computation, then make it
an optional flag.  In this case, as you correctly observe, breaking
backward compatibility also wouldn't be worth it.